File: src/app/types/coin.type.cs
import { IValue } from './value.interface';

export class Coin implements ICoin {
    name: string;
    ticker: string;
    searchString: string;
    purchasePrice: number;
    currentValue: number;
    //value?: IValue;
    quantity: number;
    purchaseDate: Date;

  constructor(coin: ICoin) {
    this.name = name;
      this.ticker = ticker;
      this.searchString = searchString;
      this.purchasePrice = purchasePrice;
      this.quantity = quantity;
      this.value = value;
      this.purchaseDate = date;
  }
}

File: src/app/types/coin.type.ts
import { CoinName } from './coinName.type';
import { PurchaseDetails } from './purchaseDetails.type';
import { Value } from './value.type';

// add builder class
export class Coin {
    name: CoinName;
    purchaseDetails: PurchaseDetails;
    value: Value;
    quantity: number;
    
    constructor(name: CoinName, purchaseDetails?: PurchaseDetails, quantity?: number, value?: Value){
        this.name = name;
        this.purchaseDetails = purchaseDetails;
        this.quantity = quantity;
        this.value = value;
    }
}
  